### PR TITLE
Integrate EAS and IPFS, publish metadata and onchain attestations

### PR DESCRIPTION
**Pull request by bswags**

### CODE REVIEW

File: src/lib/actions/projects.ts [hunk 1]
```diff
@@ -29,15 +31,26 @@ export const getProjects = async (farcasterId: string) => {
...
       error: "Unauthorized",
     }
   }
 
+  // Create project attestation
+  const attestationId = await createProjectAttestation({
+    farcasterId: session.user.farcasterId,
+  })
+
+  const projectId = nanoid()
```
Comment (bswags) [inline src/lib/actions/projects.ts:45]: i feel like the attestation schema should also include the project ID which is why I generated it here so we can update once the schema changes

File: src/components/projects/publish/Snapshot.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+export const Snapshot = ({ snapshot }: { snapshot: ProjectSnapshot }) => {
+  return (
+    <div className="flex items-center gap-2 border rounded-lg py-3 px-4">
+      <Image
+        alt="Checkmark"
+        src="/assets/icons/circle-check-green.svg"
+        height={20}
+        width={20}
+        className=""
```
Comment (tinaszheng) [inline src/components/projects/publish/Snapshot.tsx:18]: Delet

File: src/lib/actions/projects.ts [hunk 3]
```diff
@@ -29,15 +31,26 @@ export const getProjects = async (farcasterId: string) => {
 export const createNewProject = async (details: CreateProjectParams) => {
   const session = await auth()
 
-  if (!session?.user?.id) {
+  if (!session?.user?.id || !session.user.farcasterId) {
```
Comment (tinaszheng) [inline src/lib/actions/projects.ts:34]: Oh should we just have the projection created under id here for consistency?

File: src/lib/actions/snapshots.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    const { team, ...rest } = project
+    const ipfsHash = await uploadToPinata(projectId, rest)
+
+    // Create attestation
+    // TODO: Real pinata gateway URL
+    const attestationId = await createProjectMetadataAttestation({
+      projectId: project.id,
+      name: project.name,
+      category: project.category ?? "",
+      ipfsUrl: `https://op-atlas-test.mypinata.cloud/ipfs/${ipfsHash}`,
```
Comment (tinaszheng) [inline src/lib/actions/snapshots.ts:45]: Env var here perhaps

**Review by tinaszheng — APPROVED** [review]
So good

File: src/components/projects/publish/Snapshot.tsx (hunk 2 above)
Comment (bswags) [reply src/components/projects/publish/Snapshot.tsx:18]: 🔪

File: src/lib/actions/snapshots.ts (hunk 4 above)
Comment (bswags) [reply src/lib/actions/snapshots.ts:45]: ooo ya will do when we get a final one

File: src/lib/actions/projects.ts (hunk 3 above)
Comment (bswags) [reply src/lib/actions/projects.ts:34]: 🫡 will do in a follow-up

**Pull request merged**